components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/model/OpenAPIRequest.java

```diff
@@ -72,8 +87,15 @@ public OpenAPIRequest(MessageContext messageContext) {
                 .stream()
                 .collect(Collectors.toMap(Map.Entry::getKey, entry -> Collections.singleton(entry.getValue())));
         //Set transport headers
+        List<String> possibleContentTypeHeaders = new ArrayList();
+        possibleContentTypeHeaders.add("Content-Type");
+        possibleContentTypeHeaders.add("content-type");
+        possibleContentTypeHeaders.add("CONTENT-TYPE");
```

Comment:
why do we need to do this?

Reply:
possible content type headers that we need to consider and change to the camel case.

Reply:
you can just make it lowercase and check right?

Reply:
Fixed with b472b5ce30cb808ed2b02f46e03572270dfd63d6.
Thanks